pkg/ddl/partition.go

```diff
@@ -1314,16 +1316,32 @@ func buildListPartitionDefinitions(ctx sessionctx.Context, defs []*ast.Partition
 		}
 
 		buf := new(bytes.Buffer)
-		for _, vs := range clause.Values {
+		for valIdx, vs := range clause.Values {
 			inValue := make([]string, 0, len(vs))
-			for i := range vs {
-				vs[i].Accept(exprChecker)
-				if exprChecker.err != nil {
-					return nil, exprChecker.err
+			isDefault := false
+			if len(vs) == 1 {
+				if _, ok := vs[0].(*ast.DefaultExpr); ok {
+					isDefault = true
+				}
+			}
+			if len(partVals) > valIdx && len(tbInfo.Partition.Columns) > 0 && !isDefault {
```

Comment:
When is `len(partVals) > valIdx` not true when `len(tbInfo.Partition.Columns)` is?

Reply:
`len(tbInfo.Partition.Columns) > 0` is redundant. I'll only keep `len(partVals) > valIdx && !isDefault`. :+1: .